### PR TITLE
test: handle pam_nologin during connection attempt

### PR DESCRIPTION
**Pull request by MrBatschner**
**What this PR does / why we need it**:

In some rare cases, the Garden Linux platform test framework attempts to connect to a machine during its boot-up in that small time frame when SSH already accepts connections but PAM refuses them with the "System is booting up" message. In this case, the tests should not be aborted with an error but the connection should be retried.

Along the lines, a little code cleanup.

**Release note**:

```feature user
Garden Linux platform tests no longer abort on pam_nologin but attempt to retry.
```

### CODE REVIEW

File: tests/helper/sshclient.py [hunk 1]
```diff
@@ -89,6 +93,17 @@ def __init__(
         self.ssh_key_filepath = path.expanduser(sshconfig['ssh_key_filepath'])
 
 
+    def _increase_retry_count(self, wait: bool = True):
+        if self.retry_count >= self.ssh_max_retries:
+            max_timeout = self.retry_count * self.ssh_retry_timeout_seconds
+            # FIXME: this should not be pytest.exit() as it will exit immediately without cleanup
+            pytest.exit(f"Unable to establish an SSH connection after {max_timeout=} seconds. Aborting all tests.", returncode=5)
```
Comment (Vincinator) [inline tests/helper/sshclient.py:100]: Nit pick: do we need to support the case where `wait` param is set to false? 

In the wait=False case we would not do a `time.sleep(self.ssh_retry_timeout_seconds)`, but after `self.retry_count >= self.ssh_max_retries` retries, we exit with the message that we have waited for `max_timeout`

Comment (Vincinator) [reply tests/helper/sshclient.py:100]: If we remove the wait param, we should reconsider the name. 

before it was clear that this function has the side effect of a sleep. But without a param I would prefer to have it in the function name e.g.`_increase_retry_with_wait`

**Comment by Vincinator** [comment]
Fixes #1842

**Comment by MrBatschner** [comment]
Failing CI is unrelated to this change, merging.

**Pull request merged**